CriteoPublisherSdk/Tests/IntegrationTests/CR_DfpRewardedFunctionalTests.m

```diff
@@ -43,26 +46,82 @@ @implementation CR_DfpRewardedFunctionalTests
 
 - (void)setUp {
   // WARNING AdUnitHelper does use singleton object (no DI in here) should be changed with DPP-3734
-  [Criteo.sharedCriteo.dependencyProvider.integrationRegistry declare:CR_IntegrationGamAppBidding];
+  self.oldIntegrationRegistry = Criteo.sharedCriteo.dependencyProvider.integrationRegistry;
 }
 
 - (void)tearDown {
-  [Criteo.sharedCriteo.dependencyProvider.integrationRegistry declare:CR_IntegrationFallback];
+  Criteo.sharedCriteo.dependencyProvider.integrationRegistry = self.oldIntegrationRegistry;
 }
 
-- (void)test_givenRewarded_whenLoad_thenFailToReceiveAd {
+- (void)test_givenRewarded_noIntegrationDeclared_whenLoadThreeTimes_adLoaded {
   CRRewardedAdUnit *adUnit = [CR_TestAdUnits rewarded];
 
   self.criteo = [Criteo testing_criteoWithNetworkCaptor];
+  // use same integrationRegistry in test criteo object and global instance (see DPP-3734)
+  CR_IntegrationRegistry *integrationRegistry = [[CR_IntegrationRegistry alloc] init];
+  self.criteo.dependencyProvider.integrationRegistry = integrationRegistry;
+  Criteo.sharedCriteo.dependencyProvider.integrationRegistry = integrationRegistry;
+
+  // register here fails as integration is not yet detected
   [self.criteo testing_registerWithAdUnits:@[ adUnit ]];
 
   GAMRequest *request = [[GAMRequest alloc] init];
 
+  // this one make the detection logic discover GAM integration
+  [self enrichAdObject:request forAdUnit:adUnit];
+
+  // register is now ok due to GAM being detected
+  [self.criteo testing_registerWithAdUnits:@[ adUnit ]];
```

Comment:
This operation should do nothing as the criteo instance was already registered. Or maybe am I missing something ?

Reply:
Useless operation, removed it.